pkg/client/leaderelection/resourcelock/configmaplock.go

```diff
@@ -35,7 +35,7 @@ type ConfigMapLock struct {
 	// ConfigMapMeta should contain a Name and a Namespace of an
 	// ConfigMapMeta object that the Leadercmlector will attempt to lead.
 	ConfigMapMeta metav1.ObjectMeta
-	Client        clientset.Interface
+	Client        corev1client.ConfigMapsGetter
```

Comment:
> we should start using client-go, then these types will be unique

I don't think so, client-go doesn't include any federation code.

I saw this error message, it's saying federation corev1 cannot be used as clientset's ConfigMapsGetter, because although the two definitions of ConfigMapInterface are the same, to the compiler, they are two different types.

However, if you pass federation.ConfigMapInterface directly, compiler should accept it as a clientset.ConfigMapInterface, because they have the same methods defined.

I didn't try it though, so i could be wrong :)